src/main/java/org/broadinstitute/hellbender/tools/exome/GetBayesianHetCoverage.java

```diff
@@ -282,42 +285,18 @@ protected Object doWork() {
         }
 
         /* determine the job type */
-        HetCoverageJobType jobType = HetCoverageJobType.NONE;
         if (normalBamFile != null && tumorBamFile != null) {
-            jobType = HetCoverageJobType.MATCHED_NORMAL_TUMOR;
             logger.info("MATCHED_NORMAL_TUMOR mode selected.");
-        }
-        if (normalBamFile != null && tumorBamFile == null) {
-            jobType = HetCoverageJobType.NORMAL_ONLY;
+            runMatchedNormalTumor();
+        } else if (normalBamFile != null && tumorBamFile == null) {
             logger.info("NORMAL_ONLY mode selected.");
-        }
-        if (normalBamFile == null && tumorBamFile != null) {
-            jobType = HetCoverageJobType.TUMOR_ONLY;
+            runNormalOnly();
+        } else if (normalBamFile == null && tumorBamFile != null) {
             logger.info("TUMOR_ONLY mode selected.");
-        }
-
-        /* run the job */
-        switch (jobType) {
-
-            case NORMAL_ONLY:
-                runNormalOnly();
-                break;
-
-            case TUMOR_ONLY:
-                runTumorOnly();
-                break;
-
-            case MATCHED_NORMAL_TUMOR:
-                runMatchedNormalTumor();
-                break;
-
-            case NONE:
-                throw new GATKException.ShouldNeverReachHereException("The job type is not properly determined from" +
-                        " the arguments. This should not happen.");
-
-            default:
-                throw new GATKException.ShouldNeverReachHereException("The job type is not properly determined from" +
-                        " the arguments. This should not happen.");
+            runTumorOnly();
+        } else {
+            throw new GATKException.ShouldNeverReachHereException("The job type is not properly determined from" +
```

Comment:
Important to note the exception type here; use UserException when is a failure from the user to provide proper arguments.